persistent-template/Database/Persist/TH.hs

```diff
@@ -154,16 +155,16 @@ persistFileWith ps fp = persistManyFileWith ps [fp]
 -- @since 2.5.4
 persistManyFileWith :: PersistSettings -> [FilePath] -> Q Exp
 persistManyFileWith ps fps = do
-    mapM_ qAddDependentFile fps
-    ss <- mapM getS fps
-    let s = T.intercalate "\n" ss -- be tolerant of the user forgetting to put a line-break at EOF.
-    parseReferences ps s
-  where
-    getS fp = do
-      h <- qRunIO $ SIO.openFile fp SIO.ReadMode
-      qRunIO $ SIO.hSetEncoding h SIO.utf8_bom
-      s <- qRunIO $ TIO.hGetContents h
-      return s
+  mapM_ qAddDependentFile fps
+  ss <- mapM (qRunIO . getFileContents) fps
+  let s = T.intercalate "\n" ss -- be tolerant of the user forgetting to put a line-break at EOF.
+  parseReferences ps s
+
+getFileContents :: FilePath -> IO Text
+getFileContents fp = do
+  h <- SIO.openFile fp SIO.ReadMode
+  SIO.hSetEncoding h SIO.utf8_bom
+  TIO.hGetContents h
```

Comment:
While we're fixing this up, it may be good to port away from `Data.Text.IO`: https://www.snoyman.com/blog/2016/12/beware-of-readfile